gourmet/gtk/recipe.py

```diff
@@ -0,0 +1,787 @@
+from gettext import gettext as _
+import os as _os
+
+from gi.repository import (
+    Gdk as _Gdk,
+    GdkPixbuf as _GdkPixbuf,
+    Gtk as _Gtk,
+)
+
+from .. import convert as _convert
+from .. import gglobals as _globals
+from ..gtk_extras.pango_buffer import PangoBuffer as _PangoBuffer
+from ..gtk_extras.dialog_extras import select_image as _select_image_dialog
+# TODO: Move this function into GTK-specific code
+from ..image_utils import bytes_to_pixbuf as _bytes_to_pixbuf
+from ..importers.importer import parse_range as _parse_range
+from ..plugin import (
+    ToolPlugin as _ToolPlugin,
+    RecEditorPlugin as _RecEditorPlugin
+)
+from ..plugin_loader import Pluggable as _Pluggable
+# NOTE: These imports are needed to register the TimeEntry and StarRating
+# widgets that are referenced in ui files.
+from ..gtk_extras.validation import TimeEntry as _TimeEntry
+from ..gtk_extras.ratingWidget import StarButton as _StarButton
+
+
+
+# TODO: test UnitConverter (and general plugin?) support
+class RecipeEditor(_Pluggable):
+
+    def __init__(self):
+        super().__init__([_ToolPlugin, _RecEditorPlugin])
+
+
+    # TODO: Eliminate dependency on RecGui
+    def run(self, parent, rec_gui, recipe=None, module=None):
+        # TODO: preferences?
```

Comment:
It is somewhat unclear what this comment refers to

Reply:
The current recipe entry UI has a menu with a preferences option. This is a reminder to look at what the preferences were managed by that option and see if any functionality needs to be replicated.